tests/conftest.py

```diff
@@ -63,6 +59,26 @@ def entity_factory():
     return EntityFactory()
 
 
+@fixture
+def epics_root(samples: Path, tmp_path: Path, mocker: MockerFixture):
+    # create an partially populated epics_root structure in a temporary folder
+    epics = tmp_path / "epics"
+    shutil.copytree(samples / "epics", epics)
+    Path.mkdir(epics / "opi", exist_ok=True)
+    Path.mkdir(epics / "epics-base")
+    Path.mkdir(epics / "ioc/config", parents=True)
+    Path.mkdir(epics / "ibek-defs")
+    Path.mkdir(epics / "runtime")
+
+    mocker.patch.object(GLOBALS, "_EPICS_ROOT", epics)
+
+    # this should not be needed - what gives?
+    os.environ["IOC"] = "/epics/ioc"
+    os.environ["RUNTIME_DIR"] = "/epics/runtime"
+
+    return epics
```

Comment:
I think `tests/samples/epics/opi/index.bob` and `tests/samples/epics/runtime/st.cmd`
 should be removed. It looks like the tests generate files on top of these and then
 compare them to the expected, which is a bit confusing. Are these just acting as
 placeholders to allow checking the directory into git?

Reply:
Ah ah. This is why the tests are failing. The OLD tests  actually put outputs in opi and runtime. These folders are in gitignore but this dirties the samples/epics folder and then epics_root blows up trying to create.

So a fresh checkout of this repo works but our old clones with files sitting in those folders blow up.

Please can you try:

git clean -fdx
pytest tests

Reply:
I've fixed this by explicityly copying only the folders I want out of samples/epics (but really the problem will go away for new clones anyway)

Reply:
I can confirm this fixed it for me